packages/firestore/src/api/database.ts

```diff
@@ -314,55 +310,19 @@ export class MemoryPersistenceProvider implements PersistenceProvider {
  */
 export class IndexedDbPersistenceProvider implements PersistenceProvider {
   enableIndexedDbPersistence(
-    firestore: Firestore,
+    firestore: FirestoreCompat,
     forceOwnership: boolean
   ): Promise<void> {
-    const onlineComponentProvider = new OnlineComponentProvider();
-    const offlineComponentProvider = new IndexedDbOfflineComponentProvider(
-      onlineComponentProvider,
-      firestore._getSettings().cacheSizeBytes,
-      forceOwnership
-    );
-    return firestore._configureClient(
-      offlineComponentProvider,
-      onlineComponentProvider
-    );
-  }
-
-  enableMultiTabIndexedDbPersistence(firestore: Firestore): Promise<void> {
-    const onlineComponentProvider = new OnlineComponentProvider();
-    const offlineComponentProvider = new MultiTabOfflineComponentProvider(
-      onlineComponentProvider,
-      firestore._getSettings().cacheSizeBytes
-    );
-    return firestore._configureClient(
-      offlineComponentProvider,
-      onlineComponentProvider
-    );
-  }
-
-  clearIndexedDbPersistence(firestore: Firestore): Promise<void> {
-    const deferred = new Deferred();
-    firestore._queue.enqueueAndForgetEvenWhileRestricted(async () => {
-      try {
-        await indexedDbClearPersistence(
-          indexedDbStoragePrefix(
-            firestore._databaseId,
-            firestore._persistenceKey
-          )
-        );
-        deferred.resolve();
-      } catch (e) {
-        deferred.reject(e);
-      }
-    });
-    return deferred.promise;
+    return enableIndexedDbPersistence(firestore, { forceOwnership });
   }
+  enableMultiTabIndexedDbPersistence = enableMultiTabIndexedDbPersistence;
```

Comment:
This seems like it's a no-op. Did you mean to change a member of `this`? Here and the next line.

Reply:
In this instance, the `this` member doesn't matter since these new functions should not rely on an associated `this` context. To make this more readable, I assigned `null` using `bind`.

Reply:
The thing I'm missing is that this looks like an assignment of a value to itself, which should do nothing and could therefore be removed. What am I missing?

Reply:
The left part points to an instance member, while the right part is the free-standing function of the same name. JavaScript can differentiate because access to instance members requires `this`.